cli/cmd/stat.go

```diff
@@ -185,6 +199,18 @@ func writeStatsToBuffer(resp *pb.StatSummaryResponse, w *tabwriter.Writer) {
 		}
 	}
 
+	for resourceType, stats := range statTables {
+
+		if reqResourceType == k8s.All {
+			printStatTable(stats, resourceType, w, maxNameLength, maxNamespaceLength)
+		} else {
+			printStatTable(stats, "", w, maxNameLength, maxNamespaceLength)
+		}
+		fmt.Fprint(w, "\n")
+	}
+}
+
+func printStatTable(stats map[string]*row, resourceType string, w *tabwriter.Writer, maxNameLength int, maxNamespaceLength int) {
 	if len(stats) == 0 {
```

Comment:
i just realized we need to move this `len(stats)` check up into `writeStatsToBuffer`, as that's where we should be checking for the presence of any stats, otherwise we panic:

```bash
$ bin/conduit stat deploy -n foo
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/runconduit/conduit/cli/cmd.renderStats(0xc42027db50, 0x1d6ea6d, 0xb, 0xc42023fd70, 0x0)
	/go/src/github.com/runconduit/conduit/cli/cmd/stat.go:129 +0x20a
github.com/runconduit/conduit/cli/cmd.requestStatsFromAPI(0x1e84ec0, 0xc42027db30, 0xc42023fd70, 0xc4202ee630, 0x1, 0x3, 0x7ffeefbfeae6)
	/go/src/github.com/runconduit/conduit/cli/cmd/stat.go:119 +0x124
```

we can move the check like this:

```diff
diff --git a/cli/cmd/stat.go b/cli/cmd/stat.go
index 79c7566..b576d9e 100644
--- a/cli/cmd/stat.go
+++ b/cli/cmd/stat.go
@@ -199,8 +199,12 @@ func writeStatsToBuffer(resp *pb.StatSummaryResponse, reqResourceType string, w
                }
        }

-       for resourceType, stats := range statTables {
+       if len(statTables) == 0 {
+               fmt.Fprintln(os.Stderr, "No traffic found.")
+               os.Exit(0)
+       }

+       for resourceType, stats := range statTables {
                if reqResourceType == k8s.All {
                        printStatTable(stats, resourceType, w, maxNameLength, maxNamespaceLength)
                } else {
@@ -211,11 +215,6 @@ func writeStatsToBuffer(resp *pb.StatSummaryResponse, reqResourceType string, w
 }

 func printStatTable(stats map[string]*row, resourceType string, w *tabwriter.Writer, maxNameLength int, maxNamespaceLength int) {
-       if len(stats) == 0 {
-               fmt.Fprintln(os.Stderr, "No traffic found.")
-               os.Exit(0)
-       }
-
        headers := make([]string, 0)
        if allNamespaces {
                headers = append(headers,
```